src/actions/actions.ts

```diff
@@ -180,21 +185,31 @@ export abstract class BaseMovement extends BaseAction {
       }
 
       for (let i = 0; i < count; i++) {
+          const firstIteration = (i === 0);
           const lastIteration = (i === count - 1);
           const temporaryResult = (recordedState.operator && lastIteration) ?
               await this.execActionForOperator(position, vimState) :
               await this.execAction           (position, vimState);
 
-          result = temporaryResult;
-
-          if (result instanceof Position) {
-            position = result;
-          } else if (isIMovement(result)) {
-            position = result.stop;
+          if (temporaryResult instanceof Position) {
+            resultPosition = temporaryResult;
+            position = temporaryResult;
```

Comment:
Why do we assign to `position` here and elsewhere? As far as I can see, `position` is never again read from.